Nowsecure/index.ts

```diff
@@ -1,135 +1,149 @@
+    case "ARM":
+      return Arch.ARM
+    default:
+      const emsg = "Unknown system architecture"
+      tl.setResult(tl.TaskResult.Failed, emsg)
+      throw new Error(emsg)
+  }
+}
 
-let scoreThreshold = tl.getInput("scoreThreshold", false);
-tl.debug("scoreThreshold: " + scoreThreshold);
+function toolName(arch: Arch, platform: tl.Platform): string {
+  // Ex: ns_linux-amd64
+  return `ns_${platformToString(platform)}-${arch}${platform === tl.Platform.Windows ? '.exe' : ''}`
+}
 
-let waitMinutes = tl.getInput("waitMinutes", false);
-tl.debug("waitMinutes: " + waitMinutes);
+function getTool(): tr.ToolRunner {
+  const platform = tl.getPlatform()
+  const arch = archFrom(tl.getVariable("Agent.OSArchitecture"))
+  const toolPath = path.join(__dirname, toolName(arch, platform));
 
-let showStatusMessages = tl.getInput("showStatusMessages", false);
-tl.debug("showStatusMessages: " + showStatusMessages);
+  if (!fs.existsSync(toolPath)) {
+    const err =
+      "Unsupported runner type. Integration currently supports darwin/arm64, windows/amd64, and linux/amd64"
+    tl.error(err)
+    tl.setResult(tl.TaskResult.Failed, err)
+    throw new Error(err)
+  }
 
-//////////////////////////////////////////////////////////////////////////
-// Find Java executable and set parameters
-//////////////////////////////////////////////////////////////////////////
-let task = JSON.parse(fs.readFileSync(path.join(__dirname, "task.json")).toString());
-let version = `${task.version.Major}.${task.version.Minor}.${task.version.Patch}`
+  chmodx(toolPath, platform)
 
-let javaPath = tl.which("java");
-if (!javaPath) {
-  onError("java is not found in the path", 1);
-}
-let java = tl.tool("java");
-let nsAPI = path.join(__dirname, "nowsecure-ci.jar");
-
-java.arg("-jar");
-java.arg(nsAPI);
-
-
-java.arg("--plugin-name");
-java.arg("azure-nowsecure-auto-security-test");
-java.arg("--plugin-version");
-java.arg(version);
-java.arg("--file");
-java.arg(filepath);
-java.arg("--group-id");
-java.arg(group);
-java.arg("--token");
-java.arg(token);
-java.arg("--dir");
-java.arg(artifactsDir);
-
-if (url) {
-  java.arg("--url");
-  java.arg(url);
+  return tl.tool(toolPath)
 }
 
-if (waitMinutes) {
-  java.arg("--wait");
-  java.arg(waitMinutes);
-} else {
-  java.arg("--wait");
-  java.arg("0");
+type Inputs = {
+  // required params
+  filepath: string;
+  group: string;
+  token: string;
+  // optional params
+  artifact_dir?: string;
+  api_host?: string;
+  ui_host?: string;
+  log_level?: string;
+  analysisType?: string;
+  minimum_score?: string;
+  polling_duration_minutes?: string;
 }
 
-if (showStatusMessages) {
-  java.arg("--show-status-messages");
-  java.arg(showStatusMessages);
-} else {
-  java.arg("--show-status-messages");
-  java.arg("true");
+function getInputs(): Inputs {
+  const inputs = {
+    filepath: tl.getPathInputRequired("binary_file", true),
+    group: tl.getInput("group", true),
+    token: tl.getInput("token", true),
```

Comment:
Mostly a nit but I was confused looking at this originally. Above this you use the `Required` variant of the getter. Can we use that here too just to be explicit?

```suggestion
    group: tl.getInputRequired("group"),
    token: tl.getInputRequired("token"),
```